lib/serialization/BinaryMiddleware.js

```diff
@@ -331,6 +332,16 @@ class BinaryMiddleware extends SerializerMiddleware {
 					}
 					break;
 				}
+				case "bigint": {
+					const value = thing.toString();
+					const len = Buffer.byteLength(value);
+					allocate(len + HEADER_SIZE + I32_SIZE);
+					writeU8(BIGINT_HEADER);
+					writeU32(len);
+					currentBuffer.write(value, currentPosition);
+					currentPosition += len;
+					break;
+				}
```

Comment:
Why it is storing as string:
- we can't use `writeBigInt64LE`, because BigInt can be longer than 64bits
- storing by bytes (uint8 or other) will require multiple/shift them in deserialization
- we can use `thing.toString(16)` to reduce size, but I doesn't bring a lot of wins, `4294967296n.toString(16)` -> `100000000`, also it will require to store sing (-/+) and will multiple on -1

so storing them as string, I think, is the best solution

Reply:
Improved, now we store small bigints as int8 (`n <= BigInt(127) && n >= BigInt(-128)`) and int32 (`n <= BigInt(2147483647) && n >= BigInt(-2147483648)`) as regular numbers